authentication/src/main/java/io/camunda/authentication/CamundaUserDetailsService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication;
+
+import io.camunda.service.UserServices;
+import io.camunda.service.search.query.SearchQueryBuilders;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import java.util.Objects;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+public class CamundaUserDetailsService implements UserDetailsService {
+  private final UserServices<UserRecord> userServices;
+
+  public CamundaUserDetailsService(final UserServices<UserRecord> userServices) {
+    this.userServices = userServices;
+  }
+
+  @Override
+  public UserDetails loadUserByUsername(final String username) throws UsernameNotFoundException {
+    final var userQuery =
+        SearchQueryBuilders.userSearchQuery(
+            fn -> fn.filter(f -> f.username(username)).page(p -> p.size(1)));
+    return userServices.search(userQuery).items().stream()
+        .filter(Objects::nonNull)
+        .findFirst()
+        .map(
+            candidate ->
+                User.builder()
+                    .username(candidate.value().username())
+                    .password(candidate.value().password())
+                    .build())
+        .orElseThrow(() -> new UsernameNotFoundException(username));
```

Comment:
❓ should this align with the error responses from the rest API? I think ultimately we do want it to align but not sure if its something we need to address within this PR

Reply:
Depends on the amount of work, It can make sense. As I checked in BasicAuthIT we get the 401 when the user/name combination is not available, but I can't find anything specific in rest-api.yaml regarding the  messages. Could you please elaborate more on this?